dev/archery/archery/integration/tester_java.py

```diff
@@ -228,10 +246,10 @@ class JavaTester(Tester):
     CONSUMER = True
     FLIGHT_SERVER = True
     FLIGHT_CLIENT = True
-    C_DATA_SCHEMA_EXPORTER = True
-    C_DATA_SCHEMA_IMPORTER = True
-    C_DATA_ARRAY_EXPORTER = True
-    C_DATA_ARRAY_IMPORTER = True
+    C_DATA_SCHEMA_EXPORTER = _enable_c_data_tests()
+    C_DATA_SCHEMA_IMPORTER = _enable_c_data_tests()
+    C_DATA_ARRAY_EXPORTER = _enable_c_data_tests()
+    C_DATA_ARRAY_IMPORTER = _enable_c_data_tests()
```

Comment:
If we want to avoid calling `_enable_c_data_tests` multiple times, we want to reuse the first result:

```suggestion
    C_DATA_SCHEMA_EXPORTER = _enable_c_data_tests()
    C_DATA_SCHEMA_IMPORTER = C_DATA_SCHEMA_EXPORTER
    C_DATA_ARRAY_EXPORTER = C_DATA_SCHEMA_EXPORTER
    C_DATA_ARRAY_IMPORTER = C_DATA_SCHEMA_EXPORTER
```